application/apps/indexer/indexer_cli/src/main.rs

```diff
@@ -1799,3 +1799,60 @@ async fn detect_messages_type(input: &Path) -> Result<bool, DltParseError> {
         )))
     }
 }
+
+// Idea is to get the file path from the user,
+// read the first chunks from the file and return String
+// 'binary' or 'text' as of now. Later we can add enum for file
+// type.
+async fn detect_file_type(file_path: &Path) -> String {
+    let chunks = fetch_starting_chunk(file_path);
+    let buffer = match chunks {
+        Ok(buffer) => buffer,
+        // TODO: add correct error handling
+        Err(error) => panic!("Error while reading chunks from file: {}", error),
+    };
+    // println!("{:?}", buffer);
+    let result: bool = is_file_binary(buffer);
+
+    return String::from("Still in progress...");
+}
+
+fn fetch_starting_chunk(file_path: &Path) -> Result<Vec<u8>>{
+    let file = File::open(file_path)?;
+    let mut file = file.take(1024);
+    let mut buffer = Vec::new();
```

Comment:
for such a small amount of data you can put the data on the stack which is much faster.
```
let mut buffer = [0; BYTES_TO_READ];
```